products/servicedirectory/api.yaml

```diff
@@ -0,0 +1,176 @@
+# Copyright 2020 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Product
+name: ServiceDirectory
+display_name: ServiceDirectory
+versions:
+  - !ruby/object:Api::Product::Version
+    name: beta
+    base_url: https://servicedirectory.googleapis.com/v1beta1/
+scopes:
+  - https://www.googleapis.com/auth/cloud-platform
+apis_required:
+  - !ruby/object:Api::Product::ApiReference
+    name: Service Directory API
+    url: https://console.cloud.google.com/apis/library/servicedirectory.googleapis.com/
+objects:
+  - !ruby/object:Api::Resource
+    name: 'Namespace'
+    base_url: '{{name}}'
+    create_url: 'projects/{{project}}/locations/{{location}}/namespaces?namespaceId={{namespace_id}}'
+    self_link: '{{name}}'
+    update_verb: :PATCH
+    update_mask: true
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Configuring a namespace': 'https://cloud.google.com/service-directory/docs/configuring-service-directory#configuring_a_namespace'
+      api: 'https://cloud.google.com/service-directory/docs/reference/rest/v1beta1/projects.locations.namespaces'
+    iam_policy: !ruby/object:Api::Resource::IamPolicy
+      exclude: false
+      parent_resource_attribute: 'name'
+      method_name_separator: ':'
+      fetch_iam_policy_verb: :POST
+      set_iam_policy_verb: :POST
+    min_version: beta
+    description: |
+      A container for `services`. Namespaces allow administrators to group services
+      together and define permissions for a collection of services.
+    parameters:
+      - !ruby/object:Api::Type::String
+        name: 'location'
+        description: |
+          The location for the Namespace.
+          A full list of valid locations can be found by running
+          `gcloud beta service-directory locations list`.
+        required: true
+        url_param_only: true
+      - !ruby/object:Api::Type::String
+        name: namespaceId
+        description: |
+          The Resource ID must be 1-63 characters long, and comply with RFC1035.
```

Comment:
Do you think people know what RFC1035 means off the top of their head? If not, it might be nice to include a description or regex here too.